scripts/lib/CIME/namelist.py

```diff
@@ -112,7 +112,7 @@
 
 # Fortran syntax regular expressions.
 # Variable names.
-FORTRAN_NAME_REGEX = re.compile(r"(^[a-z][a-z0-9_]{0,62})(\(([+-]?\d*:?){3}\))?$", re.IGNORECASE)
+FORTRAN_NAME_REGEX = re.compile(r"(^[a-z][a-z0-9_]{0,62})(\(((\s*[+-]?\d*\s*:?\s*){3},?\s*){7}\))?$", re.IGNORECASE)
```

Comment:
I could not get multi-dimensional constructs to work. I always get an error:
`dtlimit(1,2) = 5.0`
`ERROR: Error in parsing namelist: expected literal value, but got 'dtlimit(1' at line 13, column 9`
BTW, where do the line numbers come from? This line is a comment in my user_nl_datm file and there are not 13 data entries.